lib/loglevel.js

```diff
@@ -245,12 +252,31 @@
           self.setLevel(self.levels.SILENT, persist);
       };
 
+      self.rebuild = function (includeChildren) {
+          // NOTE: includeChildren should default to `true` in v2.
```

Comment:
Ok, it sounds like this optional argument exists, and this comment, so that this can be called directly for recursive use, and within setLevel for non-recursive use.

I think it would be better to:

* Move the non-recursive logic below into a separate local function, called by both `setLevel` and `rebuild`, so that `setLevel` no longer calls `rebuild` at all.
* Drop the `includeChildren` argument - keep it simple, `rebuild()` always rebuilds children. If there's ever a use case to not do that we can investigate it later but I don't really see when it would be desirable. Since that now only affects the new rebuild API, there's no breaking change.

Reply:
Sounds good, will do.

Reply:
OK, this should be fixed in 45396397b87dacc20a62bc9bf303a70c61854be1. I also rebased this PR since #190 landed, which required some very minor test changes.

One quick question this brought up for me: should the new private method here just get merged together with `replaceLoggingMethods()`? We now have:

```js
    // Expects to be called with apply/call to set `this` to the logger.
    function replaceLoggingMethods(level, loggerName) {
        /*jshint validthis:true */
        for (var i = 0; i < logMethods.length; i++) {
            var methodName = logMethods[i];
            this[methodName] = (i < level) ?
                noop :
                this.methodFactory(methodName, level, loggerName);
        }

        // Define log.log as an alias for log.debug
        this.log = this.debug;
    }

      // Private to each logger; uses `self` instead of `this`
      function rebuildLoggingMethods() {
          var level = self.getLevel();

          replaceLoggingMethods.call(self, level, name);
          if (typeof console === undefinedType && level < self.levels.SILENT) {
              return "No console available for logging";
          }
      }
```

It doesn’t seem obvious to me that these shouldn’t just be one function; I don’t feel like there’s much responsibility difference between them other than `rebuildLoggingMethods()` checking whether `console` is present, but I’m not sure it would be bad if `replaceLoggingMethods` did that). The only other callsite for `replaceLoggingMethods()` is in `enableLoggingWhenConsoleArrives()`, and I think it *might* be better/simpler for that to be calling `rebuildLoggingMethods()` anyway.

Reply:
I don't have a strong preference here, whichever you prefer is fine by me.